django/db/models/sql/query.py

```diff
@@ -2423,9 +2428,9 @@ def annotation_select(self):
             return {}
         elif self.annotation_select_mask is not None:
             self._annotation_select_cache = {
-                k: v
-                for k, v in self.annotations.items()
-                if k in self.annotation_select_mask
+                k: self.annotations[k]
+                for k in self.annotation_select_mask
+                if k in self.annotations
```

Comment:
This change seems to be crucial as `test_union_multiple_models_with_values_list_and_annotations` still works when reverting other changes in `Query` :thinking: . Do we need them? Can we add extra tests to cover them?

Reply:
> `test_union_multiple_models_with_values_list_and_annotations` still works when reverting other changes

It does? 🤔 
Using Python 3.10.4 and 3.11.1, it fails if we keep the `set` in `self.annotation_select_mask`.
But I guess the order could be correct in the `set` by "chance".

`aggregation_regress.tests.AggregationTests.test_values_list_annotation_args_ordering` also fails locally when reverting other changes.